### DISCUSSION TITLE
Request to allow locally defined values for prop name="asset-type"

### DISCUSSION THREAD

**Discussion by telosBA** [usnistgov/OSCAL#1551] [Ideas]
Allowed values for <inventory-item> prop name="asset-type" are currently restricted to the following values:
![image](https://user-images.githubusercontent.com/104856767/201778854-c5e312c3-f459-493c-8ba3-4a64a53ef1d1.png)

We have use cases in which inventory items will be created for flashdrives, printers, services (applications), laptops, handhelds, mobile (tablets and phones), thin clients, etc... Instead of grouping these items into a predefined asset-type, we are requesting that the prop supports locally defined values.

**Comment by GaryGapinski** [comment]
Note the **must** requisite reported in this issue as well as [here](https://pages.nist.gov/OSCAL/reference/latest/system-security-plan/xml-reference/#/system-security-plan/system-implementation/inventory-item).

This issue is an apt observation of an inappropriate restriction. It is neither possible nor desirable to attempt to itemize **and constrain** all manner of asset types found in contemporary information systems.

**Reply by aj-stein-nist** [reply]
So, if I understand your point, you mean that [using `@ns` for custom `asset-type` props](https://pages.nist.gov/OSCAL/learn/tutorials/general/extension/#extending-existing-prop-values) and NIST defensively controlling the default namespace is not desirable for all manner of asset types?

**Reply by GaryGapinski** [reply]
The NIST documentation _disallows_ any (asset) types not enumerated in the NIST documentation. That seems inappropriate. The asset types provided are common but by no means comprehensive.

**Reply by aj-stein-nist** [reply]
Hi @telosBA, sorry I have not circled around to you (although I do enjoy Gary's feedback almost always): did I understand your question correctly? I believe you met with some NIST folk recently, was this topic covered and you are satisfied with how things work as-is (add with a custom `ns` on the prop) or you still want to advocate more for improvement? Let us know!

**Comment by GaryGapinski** [comment]
If using `@ns` allows other than the values specified, the XPath-like statement  "_ALLOWED VALUES for `prop[@name='asset-type]/@value`_" lacks a predicate (`[not(@ns)]`) which would render it more precise. Similarly, the statement "The value **must** be one of the following:" could be augmented with "(unless `@ns` is employed)".

The related (for `inventory-item` — there is also one for `component`)  [constraint](https://github.com/usnistgov/OSCAL/blob/d3a2b990e24210c253642451e30ea6db99bd045b/src/metaschema/oscal_implementation-common_metaschema.xml#L566-L568) does not employ `allow-other="yes"`. I am uncertain if the presence of `@ns` affects that constraint.
